provider/datagen/src/transform/cldr/datetime/week_data.rs

```diff
@@ -131,3 +133,193 @@ fn basic_cldr_week_data() {
         gg_week_data.get().first_weekday
     );
 }
+
+impl DataProvider<WeekDataV2Marker> for crate::DatagenProvider {
+    fn load(&self, req: DataRequest) -> Result<DataResponse<WeekDataV2Marker>, DataError> {
+        self.check_req::<WeekDataV2Marker>(req)?;
+        let territory = req
+            .locale
+            .region()
+            .map(|v| -> Result<Territory, DataError> { Ok(Territory::Region(v)) })
+            .transpose()?
+            .unwrap_or_else(|| DEFAULT_TERRITORY.clone());
+
+        let week_data: &cldr_serde::week_data::Resource = self
+            .cldr()?
+            .core()
+            .read_and_parse("supplemental/weekData.json")?;
+        let week_data = &week_data.supplemental.week_data;
+
+        let first_weekday: icu_calendar::types::IsoWeekday = week_data
+            .first_day
+            .get(&territory)
+            .or_else(|| week_data.first_day.get(&DEFAULT_TERRITORY))
+            .ok_or(DataError::custom(
+                "Missing default entry for firstDay in weekData.json",
+            ))?
+            .into();
+
+        let min_week_days: u8 = week_data
+            .min_days
+            .get(&territory)
+            .or_else(|| week_data.min_days.get(&DEFAULT_TERRITORY))
+            .ok_or(DataError::custom(
+                "Missing default entry for minDays in weekData.json",
+            ))?
+            .0;
+
+        let weekend: WeekendSet = {
+            let weekend_start = week_data
+                .weekend_start
+                .get(&territory)
+                .or_else(|| week_data.weekend_start.get(&DEFAULT_TERRITORY))
+                .ok_or(DataError::custom(
+                    "Missing default entry for weekendStart in weekData.json",
+                ))?;
+
+            let weekend_end = week_data
+                .weekend_end
+                .get(&territory)
+                .or_else(|| week_data.weekend_end.get(&DEFAULT_TERRITORY))
+                .ok_or(DataError::custom(
+                    "Missing default entry for weekendEnd in weekData.json",
+                ))?;
+
+            WeekendSet(weekend_start.bit_value() | weekend_end.bit_value())
```

Comment:
This only works for one and two day weekends, but you need the whole range between start and end.

You probably want to encapsulate this logic into a constructor like Zibi suggested (make sure to handle Sun/Mon weekends correctly).

Reply:
Because of the current [CLDR data](https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/weekData.json) and [Spec](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Patterns_Week_Elements), this is not a range, but discrete days. See [this example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo#obtaining_the_week_information) where `Brunei`'s weekend is represented with `Fri | Sun`, and doesn't include Saturday.

Should we consider changing CLDR to have an array instead of `weekendStart` and `weekendEnd` ?. 
By now made the parsing code use the 2-day system, but the reading and iterator code to assume it can be multiple days.

Logic is now encapsulated and documented in https://github.com/unicode-org/icu4x/pull/4364/commits/db654ead02b555c34aba8927df4956c2fd7d0c9e.